charts/consul/test/unit/cni-daemonset.bats

```diff
@@ -295,3 +295,48 @@ rollingUpdate:
       [ "${actual}" = '{"mountPath":"bar","name":"cni-net-dir"}' ]
 }
 
+@test "cni/DaemonSet: cni namespace has a default when not set" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/cni-daemonset.yaml  \
+      --set 'connectInject.cni.enabled=true' \
+      --set 'connectInject.enabled=true' \
+      . | tee /dev/stderr |
+      yq -r -c '.metadata.namespace' | tee /dev/stderr)
+  [[ "${actual}" == "default" ]]
+}
+
+@test "cni/DaemonSet: able to set cni namespace" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/cni-daemonset.yaml  \
+      --set 'connectInject.cni.enabled=true' \
+      --set 'connectInject.cni.namespace=kube-system' \
+      --set 'connectInject.enabled=true' \
+      . | tee /dev/stderr |
+      yq -r -c '.metadata.namespace' | tee /dev/stderr)
+  [[ "${actual}" == "kube-system" ]]
+}
+
+@test "cni/DaemonSet: still uses cni.namespace when helm -n is used" {
```

Comment:
I added the following two tests mostly as a sanity check that `helm -n` doesn't mess with the templating.
That being said I don't think it really is necessary to be attached to the rest of the tests since it's really just testing that `.Release.namespace` can be changed more than the templating logic.